src/base_client.py

```diff
@@ -0,0 +1,9 @@
+class BaseShellClient:
```

Comment:
You implemented BaseShellClient class to define common interface, but SSHClient and WinRMClient are not inherited from base class. Please fix